lib/ae_mdw/channels.ex

```diff
@@ -3,34 +3,149 @@ defmodule AeMdw.Channels do
   Main channels module.
   """
 
+  alias :aeser_api_encoder, as: Enc
+  alias AeMdw.Blocks
   alias AeMdw.Collection
   alias AeMdw.Db.ChannelCloseMutation
   alias AeMdw.Db.ChannelOpenMutation
   alias AeMdw.Db.ChannelSpendMutation
+  alias AeMdw.Db.ChannelUpdateMutation
   alias AeMdw.Db.Model
+  alias AeMdw.Db.Mutation
   alias AeMdw.Db.State
   alias AeMdw.Node
   alias AeMdw.Txs
+  alias AeMdw.Util
+  alias AeMdw.Validate
 
   require Model
 
   @typep state() :: State.t()
-  @type closing_type() ::
-          :channel_close_solo_tx
-          | :channel_close_mutual_tx
-          | :channel_settle_tx
+  @typep pagination() :: Collection.direction_limit()
+  @typep range() :: {:gen, Range.t()} | nil
+  @typep cursor() :: binary() | nil
+  @typep bi_txi() :: Blocks.bi_txi()
 
-  @spec close_mutation(closing_type(), Node.aetx()) :: ChannelCloseMutation.t()
-  def close_mutation(tx_type, tx), do: ChannelCloseMutation.new(tx_type, tx)
+  @type channel() :: map()
 
-  @spec open_mutation(Node.aetx()) :: ChannelOpenMutation.t()
-  def open_mutation(tx), do: ChannelOpenMutation.new(tx)
+  @table_active Model.ActiveChannel
+  @table_activation Model.ActiveChannelActivation
 
-  @spec deposit_mutation(Node.aetx()) :: ChannelSpendMutation.t()
-  def deposit_mutation(tx), do: ChannelSpendMutation.new(:aesc_deposit_tx.amount(tx))
+  @spec fetch_active_channels(state(), pagination(), range(), cursor()) ::
+          {cursor(), [channel()], cursor()}
+  def fetch_active_channels(state, pagination, range, cursor) do
+    cursor = deserialize_cursor(cursor)
+    scope = deserialize_scope(range)
 
-  @spec withdraw_mutation(Node.aetx()) :: ChannelSpendMutation.t()
-  def withdraw_mutation(tx), do: ChannelSpendMutation.new(-:aesc_withdraw_tx.amount(tx))
+    {prev_cursor, expiration_keys, next_cursor} =
+      state
+      |> build_streamer(scope, cursor)
+      |> Collection.paginate(pagination)
+
+    channels = render_list(state, expiration_keys)
+
+    {serialize_cursor(prev_cursor), channels, serialize_cursor(next_cursor)}
+  end
+
+  @spec close_mutual_mutations(bi_txi(), Node.aetx()) :: [Mutation.t()]
+  def close_mutual_mutations(bi_txi, tx) do
```

Comment:
I've used `AeMdw.Db.Channels` since we have used it before as well and I don't think we should distinguish between so many modules for a single resource `AeMdw.Channels` and `AeMdw.Db.Channels` should be enough

Reply:
I think once `AeMdw.Db.Channels` doesn't exist the mutation factory functions could be `AeMdw.Db.Sync.Channels` like `AeMdw.Db.Sync.Name` to make it clear that it exports functions to be used directly during the syncing process. But no problem, there are other modules that need to be reviewed with same goal and fits well to a specific discussion on the `db/sync` directory.

Reply:
We should def discuss it to align goals